File: kth_largest_factor_of_N.py
if __name__ == "__main__":
    a, b = map(int,input("Enter value of N, k : ").split())
    arr = []
    for i in range(1,(a+1)):
        if (a%i==0):
            arr.append(i)
    try:
        print(arr[-b])
    except:
        print(arr[0])

File: kth_largest_factor_of_N.c
#include<stdio.h>
// kth_largest_factor_of_N
int main()
{
    int n, k;
    printf("Enter N, k : ");
    scanf("%d %d", &n, &k);
    int arr[n];
    int count = 0;
    for(int i=0; i<n; i++)
    {
        while(count<n)
        {
            if(n%(i+1)==0)
            {
                arr[count]=(i+1);
                count = count + 1;
            }
            break;
            // else
            // {
            //     break;
            // }
        }
    }
    int farr[count];
    for(int j=0; j<count; j++)
    {
        int temp = arr[j];
        farr[j]=temp;
    }
    int length = sizeof(farr)/sizeof(farr[0]);
    printf("%d  ", length);
    printf("%d  ", farr[count-1]);
    return 0;
}